pool.go

```diff
@@ -110,12 +122,19 @@ func (p *pool) doRefresh() {
 	newEP, err := p.loadPool()
 	if err == nil {
 		p.lk.RLock()
+
+		// TODO: The orchestrator periodically prunes "bad" L1s based on a reputation system
+		// it owns and runs. We should probably just forget about the Saturn endpoints that were
+		// previously in the pool but are no longer being returned by the orchestrator. It's highly
+		// likely that the Orchestrator has deemed them to be non-functional/malicious.
+		// Let's just override the old pool with the new endpoints returned here.
```

Comment:
@willscott @lidel 

The orchestrator periodically tests the L1s to mantain a list of responsive L1s.

Reply:
I've asked for access to https://github.com/filecoin-saturn/orchestrator to see what exactly is "checked".

I don't think we can trust Orchestrator to give us usefull L1s, we need to verify ourselves.

The tests they perform are not end-to-end, use hardcoded CIDs, and if the first try fails, they allow cached result (so nodes that lost retrieval capability, but have test CIDs in cache are never penalized):
- https://github.com/filecoin-saturn/orchestrator/blob/04b1f712ef612b340142f3e49e89a995e7e3154e/cron/health-check.js#L27
- https://github.com/filecoin-saturn/orchestrator/blob/04b1f712ef612b340142f3e49e89a995e7e3154e/cron/random-health-check.js#L20-L32

Orchestrator opinion would be useful only if it would run a real IPFS node, add random data to IPFS, announce it on DHT, and then attempt to retrieve it via specific L1. 

This means "nearby" has little value. 
We need to fetch as many L1s we can, and then let the weighting logic to find the most useful ones.
Currently, we only get 25, but with https://github.com/filecoin-saturn/orchestrator/issues/83 we will be able to ask for more.

Reply:
@lidel This make sense to me. I think you'll have to make this change in Bifrost GW as caboose uses whatever orchestrator API URL you pass to it. Caboose Config param you'll have to change is called `OrchestratorEndpoint`.

Reply:
i think mostly this was: what happens if we get an empty list from the orchestrator because it has a bug / looses it's DB?
it doesn't seem to hurt much to keep using known, working nodes.
maybe we ask for some way for the orchestrator to more forcefully tell us to replace?

Reply:
We could be asking for `?count=9999` and see all of them, and decrease weight of ones that were removed.
This way if  they return empty list, of only 4 for some reason, we are not left with broken client – weights will be applied to all, so no impact.